src/Dapr.AspNetCore/DaprEndpointRouteBuilderExtensions.cs

```diff
@@ -33,14 +38,44 @@ public static IEndpointConventionBuilder MapSubscribeHandler(this IEndpointRoute
             {
                 var dataSource = context.RequestServices.GetRequiredService<EndpointDataSource>();
                 var entries = dataSource.Endpoints
-                    .Select(e => e.Metadata.GetMetadata<TopicAttribute>()?.Name)
-                    .Where(n => n != null)
+                    .OfType<RouteEndpoint>()
+                    .Where(e => e.Metadata.GetMetadata<TopicAttribute>()?.Name != null)   // only endpoints which have  TopicAttribute with not null Name.
                     .Distinct()
-                    .ToArray();
+                    .Select(e => (e.Metadata.GetMetadata<TopicAttribute>().Name, e.RoutePattern));
 
                 context.Response.ContentType = "application/json";
-                await JsonSerializer.SerializeAsync(context.Response.Body, entries, context.RequestServices.GetService<JsonSerializerOptions>());
+                using Utf8JsonWriter writer = new Utf8JsonWriter(context.Response.BodyWriter);
+                writer.WriteStartArray();
+
+                var logger = context.RequestServices.GetService<ILogger<TopicAttribute>>();
```

Comment:
The type is only used as the category name of the logger - by convention categories look like type names. You can also resolve `ILoggerFactory` and create a logger with an arbitrary name. This seems fine, a small percentage of people will see this.

Reply:
Sounds good, changing it to ILoggerFactory